### PR TITLE
API: Add Array API aliases (math, bitwise, linalg, misc) [Array API]

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @rgommers @ngoldbaum,

This PR adds Array API compatible functions that are direct aliases of existing functions:
- trigonometry: `acos, acosh, asin, asinh, atan, atanh, atan2`
- bitwise: `bitwise_left_shift, bitwise_invert, bitwise_right_shift`
- linalg: `tensordot, matmul`
- misc: `concat, permute_dims, pow`

Tracking issue: https://github.com/numpy/numpy/issues/25076

### CODE REVIEW

File: doc/source/reference/routines.array-manipulation.rst [hunk 1]
```diff
@@ -32,6 +32,7 @@ Transpose-like operations
    swapaxes
    ndarray.T
    transpose
+   premute_dims
```
Comment (eendebakpt) [inline doc/source/reference/routines.array-manipulation.rst:35]:
```suggestion
   permute_dims
```

File: numpy/tests/test_public_api.py [hunk 2]
```diff
@@ -605,10 +605,22 @@ def test_functions_single_location():
...
+                            "left_shift",  # np.bitwise_left_shift
+                            "right_shift",  # np.bitwise_right_shift
                             "conjugate",  # np.conj
-                            "invert",  # np.bitwise_not
+                            "invert",  # np.bitwise_not & np.bitwise_invert
                             "remainder",  # np.mod
                             "divide",  # np.true_divide
+                            "concatenate",  # np.concat
+                            "power",  # np.pow
+                            "transpose",  # np.premute_dims
```
Comment (eendebakpt) [inline numpy/tests/test_public_api.py:623]:
```suggestion
                            "transpose",  # np.permute_dims
```

Comment (mtsokol) [reply numpy/tests/test_public_api.py:623]: Uh a typo, thanks!

File: doc/source/reference/routines.array-manipulation.rst (hunk 1 above)
Comment (mtsokol) [reply doc/source/reference/routines.array-manipulation.rst:35]: Thanks!

File: doc/release/upcoming_changes/25086.new_feature.rst [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Array API compatible functions' aliases
+---------------------------------------
+
+13 aliases for existing functions were added to comply with Array API:
```
Comment (ngoldbaum) [inline doc/release/upcoming_changes/25086.new_feature.rst:4]:
```suggestion
13 aliases for existing functions were added to improve compatibility with the Array API standard:
```

**Review by ngoldbaum — COMMENTED** [review]
Just a language tweak for the release note

File: doc/release/upcoming_changes/25086.new_feature.rst (hunk 3 above)
Comment (mtsokol) [reply doc/release/upcoming_changes/25086.new_feature.rst]: Done!

**Comment by ngoldbaum** [comment]
Looks good to me. Since this has only been open for two days and it's adding new things to the public API I'm going to wait until next week to merge this. In the meantime if anyone has any issues with adding these new aliases please let us know.

**Comment by ngoldbaum** [comment]
This one has conflicts too! Sorry about not merging this in timely manner like I said I would a month ago.

**Comment by mtsokol** [comment]
I rebased it, now CI is green.

**Comment by ngoldbaum** [comment]
OK, let's pull this one in next. Thanks @mtsokol!

**Pull request merged**